Fix bounds errors when resampling with some arbitrary ratios

anowacki opened the pull request:
In some cases when `filt(::FIRFilter{FIRArbitrary}, x)` is called
with certain values of `x`, the buffer is actually one sample too
short and a `BoundsError` is thrown in
`filt!(buffer, ::FIRFilter{FIRArbitrary}, x)`.  Add one extra sample
to the calculated buffer length to catch these exceptional cases, which
mostly arise when resampling with particular resampling ratios.

(This code is really a hack and simply works around the deeper problem
of ~calculating the correct output buffer length~ writing the correct number of points in `filt!`; this should instead be
addressed properly in a future change.)

Closes #317.

martinholters reviewed (approved):
This is an ugly hack, but the situation is certainly better than before...

anowacki commented:
The following isn't quite correct:

> (This code is really a hack and simply works around the deeper problem
of calculating the correct output buffer length; this should instead be
addressed properly in a future change.)

The output buffer is the correct length; it is the writing to the buffer in the `filt!` function which is at fault, as it occasionally writes one point too many in the stream filtering routines.  So I have updated the comment and commit message to better reflect that.

ViralBShah commented:
@anowacki Would it be helpful for you to have commit access with the repo to help with maintenance? Not a whole lot changes workflow wise (PRs, reviews, etc), but we have more hands to keep things going and to improve the package.

ViralBShah commented:
Thanks. Happy to add you later if it makes sense. Just let me know.

ViralBShah commented:
Merge?

tknopp commented:
I ran into this issue and found that it is fixed on the main branch. @martinholters would it be possible to trigger a release including this fix?
